src/state/opamSwitchState.ml

```diff
@@ -278,8 +340,75 @@ let load lock_kind gt rt switch =
       conf_files
       OpamPackage.Set.empty
   in
+  (* depext check *)
+  let sys_packages =
+    if OpamStateConfig.(!r.ignore_depext || !r.depext_ignore_opam) then
+      OpamPackage.Map.empty
+    else
+      system_packages switch_config packages
+        ~depexts:(fun package ->
+            let env =
+              OpamPackageVar.resolve_switch_raw ~package gt switch switch_config
+            in
+            depexts_raw ~env package opams)
```

Comment:
We probably can limit to `available_packages` no ?

Reply:
It would force to compute the lazy value of available packages...

Reply:
Not if `sys_packages` is lazy as well ^^

Reply:
The thing is making `sys_packages`lazy would be useful only on few cases : when `no-consistency-checks` is on and there is no (real) install → system checks are not done. As at switch load, we check that installed packages still have their dependencies installed, ftm we use `sys_packages`. One solution is to make `sys_package` lazy and to launch another system querying only for installed packages, but if we're not in the case above, we'll launch system querying twice (not for the same size of package set, ok, but it remains twice).

Reply:
Well, if we want to keep `available_packages` (and upcoming `invalidated`) as lazy, it makes little sense to have `sys_packages` — which is potentially much more costly — not lazy in my opinion. Another reason is that it can pop messages, which you don't really want to see when you are doing non-action (e.g. `install`, `remove`...) commands; although we are putting effects into lazy here, which isn't best practice :innocent: if you have a better suggestion...

About `intalled`, why not simply run the check on `available ++ installed` then ? Seems straight-forward enough...

Reply:
Lazy & on available packages! On top of switch-invariants